File: app/src/test/java/com/bitquartet/tgscheduler/app/TestNSApplication.java
package com.bitquartet.tgscheduler.app;

import android.app.AlarmManager;
import android.content.Context;

import static org.mockito.Mockito.mock;

public class TestNSApplication extends NSApplication {

    private static Context context = mock(Context.class);
    private static AlarmManager alarmManager = mock(AlarmManager.class);

    @Override
    public Context getContext() {
        return context;
    }

    @Override
    public AlarmManager getAlarmManager() {
        return alarmManager;
    }
}

File: app/src/test/java/com/bitquartet/tgscheduler/app/TestNSApplication.kt
package com.bitquartet.tgscheduler.app

import android.app.AlarmManager
import android.content.Context

import org.mockito.Mockito.mock

class TestNSApplication : NSApplication() {

    override val context: Context
        get() = mockedContext

    override val alarmManager: AlarmManager
        get()  = mockedAlarmManager

    companion object {
        private val mockedContext = mock(Context::class.java)
        private val mockedAlarmManager = mock(AlarmManager::class.java)
    }
}
